lib/badge/runner.rb

```diff
@@ -80,33 +81,35 @@ def run(path, options)
       end
     end
 
-    def add_shield(icon, result, shield, alpha_channel, shield_gravity, shield_no_resize)
+    def add_shield(icon, result, shield, alpha_channel, shield_gravity, shield_no_resize, shield_scale, shield_geometry)
       UI.message "'#{icon.path}'"
       UI.verbose "Adding shield.io image ontop of icon".blue
 
-      current_shield = MiniMagick::Image.open(shield.path)
-      
-      if icon.width > current_shield.width && !shield_no_resize
-        current_shield.resize "#{icon.width}x#{icon.height}<"
+      svg_shield = MiniMagick::Image.open(shield.path)
+      new_path = "#{shield.path}.png"
+      shield_scale = shield_scale ? shield_scale.to_f : 1.0
+
+      if shield_no_resize
+        `rsvg-convert #{shield.path} -z #{shield_scale} -o #{new_path}`
       else
-        current_shield.resize "#{icon.width}x#{icon.height}>"
+        `rsvg-convert #{shield.path} -w #{(icon.width * shield_scale).to_i} -a -o #{new_path}`
       end
-      
-      result = composite(result, current_shield, alpha_channel, shield_gravity || "north")
+
+      png_shield = MiniMagick::Image.open(new_path)
+
+      result = composite(result, png_shield, alpha_channel, shield_gravity || "north", shield_geometry)
     end
 
     def load_shield(shield_string)
-      url = Badge.shield_base_url + Badge.shield_path + shield_string + ".png"
-      file_name = shield_string + ".png"
+      url = Badge.shield_base_url + Badge.shield_path + shield_string + ".svg"
+      file_name = shield_string + ".svg"
 
       UI.verbose "Trying to load image from shield.io. Timeout: #{Badge.shield_io_timeout}s".blue
       UI.verbose "URL: #{url}".blue
 
-      shield = Tempfile.new(file_name).tap do |file|
-        file.binmode
-        file.write(open(url).read)
-        file.close
-      end
+      Curl::Easy.download(url, file_name)
```

Comment:
This would be ok ... but I am not sure if the download fails the above `rescue OpenURI::HTTPError => error` will catch the error then ... please check that